pkg/http/middleware/analytics.go

```diff
@@ -0,0 +1,161 @@
+package middleware
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"regexp"
+
+	// TODO(remove): Remove auth packages
+	"go.signoz.io/signoz/pkg/query-service/auth"
+
+	"github.com/gorilla/mux"
+	v3 "go.signoz.io/signoz/pkg/query-service/model/v3"
+	"go.signoz.io/signoz/pkg/query-service/telemetry"
+	"go.uber.org/zap"
+)
+
+type AnalyticsMiddleware struct {
```

Comment:
We dont want to expose this outside the package
```suggestion
type analytics struct {
```